core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -340,6 +385,45 @@ private[spark] class ExecutorAllocationManager(
     }
   }
 
+  /**
+   * Maximum number of executors to be removed per dra evaluation.
```

Comment:
Why is something like this not relevant for batch queries?

Reply:
It can be useful for batch jobs as well. Like allocation ratio, de-allocation ratio are also relevant in scaling-out/back only few executors at a time. Especially in cases where Stages of a batch job  has alternating high/low tasks.

I restricted this behind `streamingDRAFeatureEnabled` initially for feature distinction. I can modify the code to have this as a config which can be used in traditional DRA as well.

Reply:
It could potentially be relevant to batch queries, but the algorithm for increasing the number of executors is also supposed to help not allocate to many in the first place.  Which also isn't ideal for some workloads.  I would say if you have suggestions and data proving the some drain algorithm works better then the existing then we should make it configurable and add it.  Removing and added executors can be very application dependent.  you could remove something and then need it a few seconds later... you could have a long tail task that takes an hour longer then everything else and removing those asap is going to save you money/get better cluster utilization.  Streaming case is naturally going to be different from many batch workloads.

I do think it should be a separate pr and discussion that adds it to batch though.

Reply:
Thanks for looking into this potential feature.

Sounds fair. Will limit these configs just for streaming use-case. 

>I do think it should be a separate pr and discussion that adds it to batch though.

Sure. Maybe in future there are people requesting it for batch, we can change this.